src/metabase/api/user_key_value.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.api.user-key-value
+  (:require [compojure.core :refer [GET PUT DELETE]]
+            [malli.core :as mc]
+            [malli.experimental.time.transform :as mett]
+            [malli.transform :as mtx]
+            [metabase.api.common :as api]
+            [metabase.models.user-key-value :as user-key-value]
+            [metabase.models.user-key-value.types :as types]
+            [metabase.util.malli.schema :as ms]))
+
+(api/defendpoint PUT "/namespace/:namespace/key/:key"
```

Comment:
does this make the namespace required? The schema makes it seem optional, but that means the route would have double slashes?

Reply:
good point. I assumed namespaces were required, in which case the DB field can be non-nullable. @johnswanson do you know what the intent is?

Reply:
I think this makes a non-blank string required for the namespace. line 17 will enforce that we don't allow for the "" namespace (which would have its schema stored in resources/.edn)

Reply:
Yeah, I think it makes sense to require namespaces to start. I'll update the table schema.